protocol/txsender/sender.go

```diff
@@ -64,12 +65,9 @@ func (s *privateKeySender) SetContractAddress(_ ethcommon.Address) {}
 func (s *privateKeySender) NewSwap(_ types.Hash, _pubKeyClaim [32]byte, _pubKeyRefund [32]byte,
 	_claimer ethcommon.Address, _timeoutDuration *big.Int, _nonce *big.Int,
 	value *big.Int) (ethcommon.Hash, *ethtypes.Receipt, error) {
-	s.txOpts.Value = value
-	defer func() {
-		s.txOpts.Value = nil
-	}()
-
-	tx, err := s.contract.NewSwap(s.txOpts, _pubKeyClaim, _pubKeyRefund, _claimer, _timeoutDuration, _nonce)
+	txOpts := *s.txOpts // make a copy, so we don't modify the original
```

Comment:
The `bind.TransactOpts` object is usually created fresh for every transaction. We don't have the information to do that here, so instead I changed the code to never modify `s.txOpts` and always use a copy in transactions.  There are many fields in this object that are initialized on-demand when creating the transaction, but only if they don't already have a value set.  We were running into errors, because fees had been calculated on a different transaction and then were being reused against a transaction that cost more.